### DISCUSSION TITLE
Can't get autocompletion to show up in buffer

### DISCUSSION THREAD

**Discussion by jasontheiler** [hrsh7th/nvim-cmp#1212] [Q&A] [answered]
Hi everyone 👋 

I'm fairly new to Neovim and its ecosystem and I'm currently having trouble trying to get `nvim-cmp` to work together with [`mason-lspconfig.nvim`](https://github.com/williamboman/mason-lspconfig.nvim).

I'm on Version `0.8.0` and I use [`packer.nvim`](https://github.com/wbthomason/packer.nvim) to manage my plugins, which I do in my [`lua/plugins/init.lua`](https://github.com/jasontheiler/dotfiles/blob/main/nvim/.config/nvim/lua/plugins/init.lua) file. The order of the plugins should be correct, if I'm not mistaken. I then configure all plugins using `packer.nvim`'s `config` key, where I `require` another file from my [`lua/plugins/configs/`](https://github.com/jasontheiler/dotfiles/tree/main/nvim/.config/nvim/lua/plugins/configs) directory:

- In the [`lua/plugins/configs/cmp.lua`](https://github.com/jasontheiler/dotfiles/blob/main/nvim/.config/nvim/lua/plugins/configs/cmp.lua) file I set up `nvim-cmp` with minimal configuration (for now).
- In the [`lua/plugins/configs/mason-lspconfig.lua`](https://github.com/jasontheiler/dotfiles/blob/main/nvim/.config/nvim/lua/plugins/configs/mason-lspconfig.lua) file I set up `mason-lspconfig.nvim` and declare an `on_attach` function and the `capabilities`. These are then passed to the handler functions in the `mason_lspconfig.setup_handlers()` function.

I used the [recommended configuration section](https://github.com/hrsh7th/nvim-cmp#recommended-configuration) and the [Rust-specific example](https://github.com/hrsh7th/nvim-cmp/wiki/Language-Server-Specific-Samples#rust-with-rust-toolsnvim) to get to where I am right now.

When I open a Rust file (for example) the `rust_analyzer` LSP attaches correctly:

![image](https://user-images.githubusercontent.com/50220557/193627973-6d4aeb02-9f46-4d0b-ae72-0a3ad87c77f4.png)

But I can't get the autocompletion to show up by pressing `Ctrl`+`Space`, and when I execute `:CmpStatus`, I only get this:

![image](https://user-images.githubusercontent.com/50220557/193628619-8ef80273-69e6-4372-9060-b59aaa89463f.png)

I'm pretty sure that I've made a mistake somewhere. I probably overlooked or misunderstood something, but I can't figure out what...

Any help would be greatly appreciated! 🙏

**Comment by jasontheiler** [comment, marked as answer]
Okay, I found the problem by myself. I misunderstood the syntax of the [`packer.nvim`](https://github.com/wbthomason/packer.nvim) `use` function. Basically, you can't just install multiple plugins in one `use` function like this:
```lua
-- does not work!
use({
  "<package_1>",
  "<package_2>",
  config = function() ... end,
})
```
You should instead just use two separate `use` functions:
```lua
use({
  "<package_1>",
  config = function() ... end,
})
use({
  "<package_2>",
  config = function() ... end,
})
```
I got really confused because [`packer.nvim`](https://github.com/wbthomason/packer.nvim) doesn't actually throw you any errors. It made it difficult for me to notice it.

**Reply by wookayin** [reply]
Alternatively you can also do like:
```
use {
  "<package_1>", 
  requires = { "<dep_1>", "<dep_2>", { "<dep_3>", opt = true}, ... },
  config = ...
}
```
